cmd/upgrade.go

```diff
@@ -0,0 +1,133 @@
+package cmd
+
+import (
+	"archive/tar"
+	"compress/gzip"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"path/filepath"
+	"runtime"
+	"strings"
+	"time"
+
+	"github.com/coreos/go-semver/semver"
+	"github.com/spf13/cobra"
+)
+
+var upgradeCmd = &cobra.Command{
+	Use:   "upgrade",
+	Short: "Upgrade osdctl",
+	Long:  "Fetch latest osdctl from GitHub and replace the running binary",
+	RunE:  upgrade,
+}
+
+func upgrade(cmd *cobra.Command, args []string) error {
+	latest, err := getLatestVersion()
+	if err != nil {
+		return err
+	}
+	latestWithoutPrefix := strings.TrimPrefix(latest, "v")
+	// check if an upgrade is necessary
+	currentSemVer := semver.New(Version)
+	latestSemVer := semver.New(latestWithoutPrefix)
+	if currentSemVer.LessThan(*latestSemVer) {
```

Comment:
```suggestion
	if !currentSemVer.LessThan(*latestSemVer) {
	   fmt.Println("Osdctl is already in newest version")
	   return nil
	}
	...
```
removes one level of nesting

Reply:
Good catch!